phpunit/block-supports/layout-test.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Test the block layout support.
+ *
+ * @package Gutenberg
+ */
+
+class WP_Block_Supports_Layout_Test extends WP_UnitTestCase {
+	function setUp() {
+		parent::setUp();
+		$this->theme_root     = realpath( __DIR__ . '/../data/themedir1' );
+		$this->orig_theme_dir = $GLOBALS['wp_theme_directories'];
+
+		// /themes is necessary as theme.php functions assume /themes is the root if there is only one root.
+		$GLOBALS['wp_theme_directories'] = array( WP_CONTENT_DIR . '/themes', $this->theme_root );
+
+		add_filter( 'theme_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'stylesheet_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'template_root', array( $this, 'filter_set_theme_root' ) );
+		$this->queries = array();
+		// Clear caches.
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+	}
+
+	function tearDown() {
+		$GLOBALS['wp_theme_directories'] = $this->orig_theme_dir;
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+		parent::tearDown();
+	}
+
+	function filter_set_theme_root() {
+		return $this->theme_root;
+	}
+
+	function test_outer_container_restored_for_aligned_image_block_with_non_themejson_theme() {
+		// The "default" theme doesn't have theme.json support.
+		switch_theme( 'default' );
+		$block         = array(
+			'blockName' => 'core/image',
+			'attrs'     => array(
+				'className' => '',
+			),
+		);
+		$block_content = '<figure class="wp-block-image alignright size-full"><img src="/my-image.jpg"/></figure>';
+		$actual        = gutenberg_restore_image_outer_container( $block_content, $block );
+		$expected      = '<div class="wp-block-image "><figure class=" alignright size-full"><img src="/my-image.jpg"/></figure></div>';
```

Comment:
Nit: I see some extra spaces in the output, not that it's very important but I was wondering if we should account for them in the regex maybe

Reply:
Fixed